### PR TITLE
[SPARK-33892][SQL] Display char/varchar in DESC and SHOW CREATE TABLE

### PR DESCRIPTION
**Pull request by yaooqinn**
### What changes were proposed in this pull request?

Display char/varchar in
  - DESC table
  - DESC column
  - SHOW CREATE TABLE

### Why are the changes needed?

show the correct definition for users

### Does this PR introduce _any_ user-facing change?

yes, char/varchar  column's will print char/varchar instead of string
### How was this patch tested?

new tests

### CODE REVIEW

**Comment by yaooqinn** [comment]
cc @cloud-fan @maropu @HyukjinKwon thanks a lot

File: sql/core/src/main/scala/org/apache/spark/sql/execution/datasources/v2/DescribeTableExec.scala [hunk 1]
```diff
@@ -31,6 +31,9 @@ case class DescribeTableExec(
     table: Table,
     isExtended: Boolean) extends V2CommandExec {
 
+
+  override def producedAttributes: AttributeSet = outputSet
```
Comment (cloud-fan) [inline sql/core/src/main/scala/org/apache/spark/sql/execution/datasources/v2/DescribeTableExec.scala:35]: Shall we do this in `V2CommandExec`?

File: sql/core/src/test/scala/org/apache/spark/sql/CharVarcharTestSuite.scala [hunk 2]
```diff
@@ -603,6 +611,25 @@ class FileSourceCharVarcharTestSuite extends CharVarcharTestSuite with SharedSpa
       }
     }
   }
+
+  test("DESCRIBE COLUMN w/ char/varchar") {
```
Comment (cloud-fan) [inline sql/core/src/test/scala/org/apache/spark/sql/CharVarcharTestSuite.scala:615]: shall we add a TODO? eventually, v2 catalog can run these 2 commands as well.

**Review by cloud-fan — APPROVED** [review]

File: sql/core/src/test/scala/org/apache/spark/sql/CharVarcharTestSuite.scala (hunk 2 above)
Comment (yaooqinn) [reply sql/core/src/test/scala/org/apache/spark/sql/CharVarcharTestSuite.scala:615]: Any existing JIRA to track this, or I shall create one, then mark todo w/ it togeter

File: sql/core/src/main/scala/org/apache/spark/sql/execution/datasources/v2/DescribeTableExec.scala (hunk 1 above)
Comment (yaooqinn) [reply sql/core/src/main/scala/org/apache/spark/sql/execution/datasources/v2/DescribeTableExec.scala:35]: Yea, I agree. I was wondering if it is relevant to this PR.

File: sql/core/src/test/scala/org/apache/spark/sql/CharVarcharTestSuite.scala (hunk 2 above)
Comment (yaooqinn) [reply sql/core/src/test/scala/org/apache/spark/sql/CharVarcharTestSuite.scala:615]: https://issues.apache.org/jira/browse/SPARK-33875 found it

File: sql/core/src/main/scala/org/apache/spark/sql/execution/datasources/v2/DescribeTableExec.scala [hunk 3]
```diff
@@ -22,7 +22,7 @@ import scala.collection.mutable.ArrayBuffer
 
 import org.apache.spark.sql.catalyst.InternalRow
 import org.apache.spark.sql.catalyst.encoders.RowEncoder
-import org.apache.spark.sql.catalyst.expressions.{Attribute, GenericRowWithSchema}
+import org.apache.spark.sql.catalyst.expressions.{Attribute, AttributeSet, GenericRowWithSchema}
```
Comment (dongjoon-hyun) [inline sql/core/src/main/scala/org/apache/spark/sql/execution/datasources/v2/DescribeTableExec.scala:25]: Shall we revert this change on this irrelevant file?

File: sql/core/src/test/scala/org/apache/spark/sql/CharVarcharTestSuite.scala [hunk 4]
```diff
@@ -443,6 +443,14 @@ trait CharVarcharTestSuite extends QueryTest with SQLTestUtils {
         ("c1 IN (c2)", true)))
     }
   }
+
+  test("DESCRIBE TABLE w/ char/varchar") {
```
Comment (dongjoon-hyun) [inline sql/core/src/test/scala/org/apache/spark/sql/CharVarcharTestSuite.scala:447]: Please add `SPARK-33892: ` prefix.

File: sql/core/src/test/scala/org/apache/spark/sql/CharVarcharTestSuite.scala [hunk 5]
```diff
@@ -603,6 +611,26 @@ class FileSourceCharVarcharTestSuite extends CharVarcharTestSuite with SharedSpa
       }
     }
   }
+
+  // TODO: Move this test to super after SPARK-33875 implements DESCRIBE COLUMN for v2
```
Comment (dongjoon-hyun) [inline sql/core/src/test/scala/org/apache/spark/sql/CharVarcharTestSuite.scala:615]: Please make this as IDed TODO like `TODO(SPARK-33875)` and add some description about `Move this test...` to SPARK-33875's description.

**Review by dongjoon-hyun — COMMENTED** [review]
Please fix the linter error.

**Review by HyukjinKwon — APPROVED** [review]

File: sql/core/src/test/scala/org/apache/spark/sql/CharVarcharTestSuite.scala (hunk 4 above)
Comment (cloud-fan) [reply sql/core/src/test/scala/org/apache/spark/sql/CharVarcharTestSuite.scala:447]: +1

File: sql/hive/src/test/scala/org/apache/spark/sql/HiveCharVarcharTestSuite.scala [hunk 6]
```diff
@@ -41,6 +41,15 @@ class HiveCharVarcharTestSuite extends CharVarcharTestSuite with TestHiveSinglet
     }
     super.afterAll()
   }
+
+  test("SHOW CREATE TABLE AS SERDE w/ char/varchar") {
```
Comment (cloud-fan) [inline sql/hive/src/test/scala/org/apache/spark/sql/HiveCharVarcharTestSuite.scala:45]: ditto

**Comment by cloud-fan** [comment]
thanks, merging to master/3.1!
